crates/marketplace/src/service.rs

```diff
@@ -104,7 +104,13 @@ pub struct GlobalState<TYPES: NodeType> {
     pub blocks: lru::LruCache<BlockId<TYPES>, BlockInfo<TYPES>>,
 
     // registered builder states
-    pub spawned_builder_states: HashMap<BuilderStateId<TYPES>, BroadcastSender<MessageType<TYPES>>>,
+    pub spawned_builder_states: HashMap<
+        BuilderStateId<TYPES>,
+        (
+            Option<BuiltFromProposedBlock<TYPES>>,
```

Comment:
Technically it can be `None` at any point.  At the moment the only time it **should** be `None` is at initialization.  However, it would continue to operate somewhat decently even if `None` was passed more often.  The only problem with that would be that a race condition would be present in the fork test again.

I can make this comment though.  Or if you think it's important, we can also potentially replace the `Tuple` with a data structure that represents this information more clearly.

Reply:
No need to replace the data structure. A short comment sounds good!

Reply:
Addressed in 15e7c02eb16bac0821fad33f6839852be1f79fe3